src/core/public/keyboard_shortcut/key_parser.ts

```diff
@@ -0,0 +1,555 @@
+    Minus: '-',
+
+    // Arrow keys
+    ArrowUp: 'up',
+    ArrowDown: 'down',
+    ArrowLeft: 'left',
+    ArrowRight: 'right',
+
+    // Special keys
+    Space: 'space',
+    Tab: 'tab',
+    Enter: 'enter',
+    Escape: 'escape',
+    Backspace: 'backspace',
+    Delete: 'delete',
+  };
+
+  private static readonly DISPLAY_MAPPINGS: PlatformDisplayMappings = {
+    mac: {
+      ctrl: '⌃',
+      alt: '⌥',
+      shift: '⇧',
+      cmd: '⌘',
+      enter: '↵',
+      backspace: '⌫',
+      delete: '⌦',
+      tab: '⇥',
+      escape: 'Esc',
+      space: 'Space',
+      up: '↑',
+      down: '↓',
+      left: '←',
+      right: '→',
+      plus: '+',
+    },
+    other: {
+      ctrl: 'Ctrl',
+      alt: 'Alt',
+      shift: 'Shift',
+      cmd: 'Ctrl',
+      enter: '↵',
+      backspace: '⌫',
+      delete: '⌦',
+      tab: '⇥',
+      escape: 'Esc',
+      space: 'Space',
+      up: '↑',
+      down: '↓',
+      left: '←',
+      right: '→',
+      plus: '+',
+    },
+  } as const;
+
+  private keyStringCache = new Map<string, string>();
+
+  /**
+   * Normalizes a key string from shortcut definition into consistent Mac-style format
+   *
+   * Converts various input formats into a standardized Mac-style key string for consistent
+   * codebase registration. All platforms register shortcuts the same way (Mac-style) but
+   * display them appropriately for each platform. Uses caching for performance.
+   *
+   * @param keyString - Raw key combination string (e.g., "cmd+s", "alt+shift+a")
+   * @returns Normalized Mac-style key string with cmd-first modifier order
+   * @throws Error when input is invalid, null, undefined, or malformed
+   *
+   * @example
+   *
+   * // Valid key combinations using supported modifier names:
+   * parser.normalizeKeyString("cmd+s") // → "cmd+s"
+   * parser.normalizeKeyString("alt+h") // → "alt+h"
+   * parser.normalizeKeyString("shift+cmd+a") // → "cmd+shift+a"
+   * parser.normalizeKeyString("cmd+alt+shift+z") // → "cmd+alt+shift+z"
+   *
+   * // Invalid keys are rejected:
+   * parser.normalizeKeyString("f1") // → throws Error
+   * parser.normalizeKeyString("ctrl+s") // → throws Error (use "cmd+s")
+   * parser.normalizeKeyString("cmd++") // → throws Error
+   *
+   */
+  public normalizeKeyString(keyString: string): string {
+    if (this.keyStringCache.has(keyString)) {
+      return this.keyStringCache.get(keyString)!;
+    }
+    const result = this.computeNormalizedKeyString(keyString);
+    this.keyStringCache.set(keyString, result);
+
+    return result;
+  }
+
+  /**
+   * Validates the input key string for basic format and structural issues.
+   * Throws descriptive errors for invalid inputs.
+   */
+  private validateKeyString(keyString: string): void {
+    if (keyString === null || keyString === undefined) {
+      throw new Error(`Key string cannot be null or undefined`);
+    }
+
+    if (keyString === '') {
+      throw new Error(`Key string cannot be empty: "${keyString}"`);
+    }
```

Comment:
1. we can just throw "Key string cannot be empty" and not have the string interpolation. Its always going to be empty so its not helpful

2. Instead of having the above `keyString === null` || `keyString === undefined` above, why don't we just remove that condition and here we just do `if (!keyString)`? That will handle both `null`, `undefined`, and `empty string` scenario